### PR TITLE
Improve readability for failing errors in nav area

### PR DESCRIPTION
**Pull request by Duroxxigar**
Adding human readable messages to some of the failing reasons so users don't have to rely on cryptic messaging. This should help direct new users as well.

### CODE REVIEW

File: scene/3d/navigation_agent_3d.cpp [hunk 1]
```diff
@@ -192,15 +192,15 @@ Vector3 NavigationAgent3D::get_target_location() const {
 Vector3 NavigationAgent3D::get_next_location() {
 	update_navigation();
 	if (navigation_path.size() == 0) {
-		ERR_FAIL_COND_V(agent_parent == nullptr, Vector3());
+		ERR_FAIL_COND_MSG(agent_parent == nullptr,  "The agent has no parent");
```
Comment (qarmin) [inline scene/3d/navigation_agent_3d.cpp:195]: This should be `ERR_FAIL_COND_V_MSG` instead `ERR_FAIL_COND_MSG` - V means that when error happens, something must be returned, it this situation `Vector3()`

**Review by qarmin — DISMISSED** [review]

**Comment by Calinou** [comment]
The messages should also be added to the 2D counterparts.

File: scene/3d/navigation_agent_3d.cpp [hunk 2]
```diff
@@ -192,15 +192,15 @@ Vector3 NavigationAgent3D::get_target_location() const {
...
 		return agent_parent->get_global_transform().origin;
 	} else {
 		return navigation_path[nav_path_index] - Vector3(0, navigation_height_offset, 0);
 	}
 }
 
 real_t NavigationAgent3D::distance_to_target() const {
-	ERR_FAIL_COND_V(agent_parent == nullptr, 0.0);
+	ERR_FAIL_COND_MSG(agent_parent == nullptr,  "The agent has no parent");
```
Comment (Calinou) [inline scene/3d/navigation_agent_3d.cpp:203]: Add a period at the end for consistency:

```suggestion
	ERR_FAIL_COND_MSG(agent_parent == nullptr,  "The navigation agent has no parent.");
```

File: modules/navigation/navigation_mesh_generator.cpp [hunk 3]
```diff
@@ -489,7 +489,7 @@ NavigationMeshGenerator::~NavigationMeshGenerator() {
 }
 
 void NavigationMeshGenerator::bake(Ref<NavigationMesh> p_nav_mesh, Node *p_node) {
-	ERR_FAIL_COND(!p_nav_mesh.is_valid());
+	ERR_FAIL_COND_MSG(!p_nav_mesh.is_valid(), "Invalid Navigation Mesh");
```
Comment (Calinou) [inline modules/navigation/navigation_mesh_generator.cpp:492]: Add a period at the end, use sentence case:

```suggestion
	ERR_FAIL_COND_MSG(!p_nav_mesh.is_valid(), "Invalid navigation mesh.");
```

**Review by Calinou — CHANGES_REQUESTED** [review]

File: scene/3d/navigation_agent_3d.cpp (hunk 1 above)
Comment (Duroxxigar) [reply scene/3d/navigation_agent_3d.cpp]: Updated

File: scene/3d/navigation_agent_3d.cpp (hunk 2 above)
Comment (Duroxxigar) [reply scene/3d/navigation_agent_3d.cpp]: Updated

File: modules/navigation/navigation_mesh_generator.cpp (hunk 3 above)
Comment (Duroxxigar) [reply modules/navigation/navigation_mesh_generator.cpp]: Updated

**Review by Calinou — CHANGES_REQUESTED** [review]

**Review by Calinou — APPROVED** [review]

**Pull request merged**